src/map/lua/lua_baseentity.cpp

```diff
@@ -2349,21 +2333,19 @@ inline int32 CLuaBaseEntity::openSendBox(lua_State* L)
 }
 
 /************************************************************************
- *  Function: leavegame()
+ *  Function: leaveGame()
  *  Purpose : Forces a client shutdown
- *  Example : player:leavegame()
+ *  Example : player:leaveGame()
  *  Notes   : Used for logoff commands
  ************************************************************************/
 
-inline int32 CLuaBaseEntity::leavegame(lua_State* L)
+void CLuaBaseEntity::leaveGame()
 {
     TPZ_DEBUG_BREAK_IF(m_PBaseEntity == nullptr);
     TPZ_DEBUG_BREAK_IF(m_PBaseEntity->objtype != TYPE_PC);
 
-    ((CCharEntity*)m_PBaseEntity)->status = STATUS_TYPE::SHUTDOWN;
-    charutils::SendToZone((CCharEntity*)m_PBaseEntity, 1, 0);
-
-    return 0;
+    dynamic_cast<CCharEntity*>(m_PBaseEntity)->status = STATUS_TYPE::SHUTDOWN;
```

Comment:
This code is very battle-tested, so it doesn't need to be "fixed", but this would be a better way to handle this sort of thing:

```cpp
if (auto* PChar = dynamic_cast<CCharEntity*>(m_PBaseEntity))
{
    PChar ->status = STATUS_TYPE::SHUTDOWN;
    charutils::SendToZone(PChar, 1, 0); 
}
```